broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -249,45 +347,208 @@ class ServiceFabrikApiController extends FabrikBaseController {
           .invoke()
           .tap(response => logger.info('backup response ', response))
           .then(body => res
-            .status(202)
+            .status(CONST.HTTP_STATUS_CODE.ACCEPTED)
             .send(body)
           );
       });
   }
 
+  getBackupOptions(backupGuid, req) {
+    return Promise
+      .all([
+        cf.cloudController.findServicePlanByInstanceId(req.params.instance_id),
+        cf.cloudController.getOrgAndSpaceGuid(req.params.instance_id)
+      ])
+      .spread((planDetails, orgAndSpaceDetails) => {
+        const context = req.body.context || {
+          space_guid: orgAndSpaceDetails.space_guid,
+          platform: CONST.PLATFORM.CF
+        };
+        const backupOptions = {
+          guid: backupGuid,
+          instance_guid: req.params.instance_id,
+          plan_id: req.body.plan_id || planDetails.entity.unique_id,
+          service_id: req.body.service_id || this.getPlan(planDetails.entity.unique_id).service.id,
+          arguments: req.body,
+          username: req.user.name,
+          useremail: req.user.email || '',
+          context: context
+        };
+        return backupOptions;
+      });
+  }
+
+  startBackupV2(req, res) {
+    let backupStartedAt;
+    let lockedDeployment = false; // Need not unlock if checkQuota fails for parallelly triggered on-demand backup
+    req.manager.verifyFeatureSupport(CONST.OPERATION_TYPE.BACKUP);
+    const trigger = _.get(req.body, 'trigger', CONST.BACKUP.TRIGGER.ON_DEMAND);
+    let backupGuid;
+    return Promise
+      .try(() => this.checkQuota(req, trigger))
+      .then(() => utils.uuidV4())
+      .then(guid => {
+        _.set(req.body, 'trigger', trigger);
+        backupGuid = guid;
+        return this.getBackupOptions(backupGuid, req)
+          .then(backupOptions => {
+            logger.info(`Triggering backup with options: ${JSON.stringify(backupOptions)}`);
+            // Acquire read lock
+            return lockManager.lock(req.params.instance_id, {
+                lockType: CONST.ETCD.LOCK_TYPE.READ,
```

Comment:
shouldnt Lock manager should decide locktype based on operation name?

Reply:
Done!